File: 1894-MergeStringsAlternately/1894-MergeStringsAlternately.cpp
// Last updated: 1/31/2026, 2:18:58 PM
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int n = word1.size(), m = word2.size();
        string res;
        int i = 0, j = 0;
        while(i < n || j < m){
            if(i < n) res += word1[i++];
            if(j < m) res += word2[j++];
        }
        return res;
    }
};
// ------------------------------------------------------------

// class Solution {
// public:
//     string mergeAlternately(string word1, string word2) {
//         int n = word1.size(), m = word2.size();
//         string res;
//         for (int i = 0; i < n || i < m; i++) {
//             if (i < n) {
//                 res += word1[i];
//             }
//             if (i < m) {
//                 res += word2[i];
//             }
//         }
//         return res;
//     }
// };

File: 1894-MergeStringsAlternately/1894-MergeStringsAlternately.py
# Last updated: 1/31/2026, 2:18:59 PM
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        length1 = len(word1)
        length2 = len(word2)
        i = 0
        j = 0
        output = ""
        while length1 > i or length2 > j:
            if i < length1:
                output = output + word1[i]
                i += 1
            if j < length2:
                output = output + word2[j]
                j += 1
        
        return output